src/analysis/retail/evoker/preservation/normalizers/EventLinking/helpers.ts

```diff
@@ -254,10 +257,40 @@ export function getEchoAplication(event: HealEvent | ApplyBuffEvent | RefreshBuf
   }
 }
 
-export function getDreamBreathHealing(event: ApplyBuffEvent | RefreshBuffEvent) {
+export function getDreamBreathHealing(event: ApplyBuffEvent | RefreshBuffEvent | CastEvent) {
+  if (event.type === EventType.Cast) {
+    const ApplyEvent = GetRelatedEvent(event, DREAM_BREATH_CAST);
+    if (ApplyEvent) {
+      return GetRelatedEvents<HealEvent>(ApplyEvent, DREAM_BREATH);
+    }
+  }
   return GetRelatedEvents<HealEvent>(event, DREAM_BREATH);
 }
 
+export function getDreamBreathCast(event: ApplyBuffEvent | RefreshBuffEvent) {
+  const castEvent = GetRelatedEvent<EmpowerEndEvent>(event, DREAM_BREATH_CAST);
+  if (!castEvent) {
+    const stasisEvent = GetRelatedEvent(event, DREAM_BREATH_FROM_STASIS);
+    if (stasisEvent) {
+      const stasisFill = GetRelatedEvents(stasisEvent, STASIS_FILLING);
+      if (stasisFill) {
+        for (const cast of stasisFill) {
+          const stasisSpell = GetRelatedEvent(cast, STASIS);
+          if (
+            stasisSpell &&
+            stasisSpell.type === EventType.EmpowerEnd &&
+            stasisSpell.ability.name === event.ability.name
+          ) {
+            return stasisSpell;
+          }
```

Comment:
You can use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find instead of iterating